content/docs/rfcs/17/README.md

```diff
@@ -2,127 +2,138 @@
+This is in addition to sending the deletion transaction to the membership contract.
+The benefit of an off-chain slashing is that it allows real-time removal of spammers as opposed to on-chain slashing in which peers get informed with a delay,
+where the delay is due to mining the slashing transaction.
+For the group synchronization, one important security consideration is that peers MUST make sure they always use the most recent Merkle tree root in their proof generation.
+The reason is that using an old root can allow inference about the index of the user's `pk` in the membership tree hence compromising user privacy and breaking message unlinkability.
 
 ## Routing
 
-Upon the receipt of a PubSub message, the routing peer needs to extract and parse the `proof` from the `data` field.  
-If the `epoch` attached to the message has a non-reasonable gap (TODO: the gap should be defined) with the routing peer's current `epoch` then the message must be dropped (this is to prevent a newly registered peer spamming the system by messaging for all the past epochs). 
-Furthermore, the routing peers MUST check whether the `proof` is valid and the message is not spam. 
-If both checks are passed successfully, then the message is relayed. 
-If `proof` is invalid then the message is dropped. 
-If spamming is detected, the publishing peer gets slashed. 
-An overview of routing procedure is depicted in Figure 2.
-
-### Spam Detection and Slashing
-In order to enable local spam detection and slashing, routing peers MUST record the `nullifier`, `share_x`, and `share_y` of any incoming message conditioned that it is not spam and has valid proof. 
-To do so, the peer should follow the following steps. 
-1. The routing peer first verifies the `zkSNARKs` and drops the message if not verified. 
-2. Otherwise, it checks whether a message with an identical `nullifier` has already been relayed. 
-   1. If such message exists and its `share_x` and `share_y` components are different from the incoming message, then slashing takes place (if the `share_x` and `share_y` fields of the previously relayed message is identical to the incoming message, then the message is a duplicate and shall be dropped).
-   2. If none found, then the message gets relayed.
-
-An overview of slashing procedure is provided in Figure 2.
+Upon the receipt of a PubSub message via [`11/WAKU2-RELAY`](/spec/11) protocol, the routing peer parses the `data` field as a `WakuMessage` and gets access to the `RateLimitProof` field.  
+The peer then validates the `RateLimitProof`  as explained next.
 
-<!-- TODO: may shorten or delete the Spam detection and slashing process -->
+**Epoch Validation**
+If the `epoch` attached to the message is more than `Max_Epoch_Gap` apart from the routing peer's current `epoch` then the message is discarded and considered invalid.
```

Comment:
Nitpick but for constants can we keep them `lower_case`? E.g. `max_epoch_gap` (same above)

Reply:
sure done in https://github.com/vacp2p/rfc/pull/484/commits/8954039fe162421351bcacf73cba41ef864bca21